testsuite/forge/src/interface/public.rs

```diff
@@ -44,19 +45,41 @@ impl<'t> PublicUsageContext<'t> {
         self.public_info.chain_id
     }
 
+    pub fn tx_factory(&self) -> TransactionFactory {
+        self.public_info.tx_factory.clone()
+    }
+
     pub fn fund(
         &mut self,
         currency: Currency,
         auth_key: AuthenticationKey,
         amount: u64,
+        tx_factory: TransactionFactory,
+    ) -> Result<()> {
+        self.public_info
+            .coffer
+            .fund(currency, auth_key, amount, tx_factory)
+    }
```

Comment:
I don't think the fund function should be changed, there's no real reason for it and it actually ends up leaking the abstraction. specifically the "funding" can be done either through a faucet service or via the Coffer actually having the mint key but the caller shouldn't care about this implementation detail, so i would change back the fund function, trait and coffer.

Reply:
oh, this should be the display format here, all the change for fund here is one more tx_factory pass in, it will be removed since addressed other comments

Reply:
My thought is I do not really prefer let Coffer to hold a tx_factory, I more like to make tx_factory in Context, so I removed tx_factory out of Coffer::TreasuryCompliance